streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1407,6 +1430,26 @@ private void verifyMaxInFlightRequestPerConnection(final Object maxInFlightReque
         return props;
     }
 
+    public long getTotalCacheSize() {
+        // both deprecated and new config set. Warn and use the new one.
+        if (originals().containsKey(CACHE_MAX_BYTES_BUFFERING_CONFIG) && originals().containsKey(STATESTORE_CACHE_MAX_BYTES_CONFIG)) {
+            log.warn("Use of deprecated config {} noticed.", CACHE_MAX_BYTES_BUFFERING_CONFIG);
```

Comment:
This warn log line would be confusing I'm afraid, since we print this line in both cases, whereas here we are not actually using the deprecated config, but the new one. I'd suggest we differentiate this line with the line 1445 below, e.g.:

* Remove this log line in 1436.
* Update log line in 1438:

```
Both deprecated config {} and the new config {} are set, hence {} is ignored and {} is used instead.
```

* Update log line in 1446 as:

```
Deprecated config {} is set, and will be used; we suggest setting the new config {} instead as deprecated {} would be removed in the future.
```